### ISSUE TITLE
using two models, one not to be trained (frozen)

### ISSUE THREAD

**Issue by fawazsammani** [huggingface/accelerate#132]
Hi, Hope your well!
Im working on a project where I have 2 models: The first is the model that is trained, and the second is a teacher model to distil knowledge from. My training loop looks like this on a high level:

```
 for epoch in range(epochs):
    
    model.train()
    
    for step, batch in enumerate(train_loader):
       
        with torch.no_grad():
            teacher_features = teacher_model(img)
        
        base_loss, distil_loss= model(**batch, teacher_features)
        loss = base_loss + distil_loss
        accelerator.backward(loss)
```
Since each process in distributed training needs the teacher features, should the teacher model go to the `prepare` function as well? Also note that all my losses (including the distillation loss) are computed inside the model (just like huggingface models such as GPT)

**Comment by sgugger** [comment]
As long as it's not trained, it doesn't need to go in the `prepare` method (which will wrap the model in the container that all-reduce the gradients, but there is no gradient).

**Comment by fawazsammani** [comment]
I uninstalled it and re-installed it with `pip install git+https://github.com/huggingface/accelerate` and now it works
Previously i installed it with `pip install accelerate` so it doesent seem to install from main
